src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3104,11 +3108,215 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
+            }
+
+            return initializedInstance;
+        }
+
+        private void HandleObjectOrCollectionInitializer(IObjectOrCollectionInitializerOperation initializer, IOperation initializedInstance)
+        {
+            // PROTOTYPE(dataflow): Handle collection initializers
+            IOperation previousInitializedInstance = _currentInitializedInstance;
+            _currentInitializedInstance = initializedInstance;
+
+            foreach (IOperation innerInitializer in initializer.Initializers)
+            {
+                handleInitializer(innerInitializer);
+            }
+
+            _currentInitializedInstance = previousInitializedInstance;
+
+            void handleInitializer(IOperation innerInitializer)
+            {
+                switch (innerInitializer.Kind)
+                {
+                    case OperationKind.MemberInitializer:
+                        handleMemberInitializer((IMemberInitializerOperation)innerInitializer);
+                        return;
+
+                    case OperationKind.SimpleAssignment:
+                        AddStatement(handleSimpleAssignment((ISimpleAssignmentOperation)innerInitializer));
+                        return;
+
+                    case OperationKind.CollectionElementInitializer:
+                        // PROTOTYPE(dataflow): support collection initializers
+                        throw new NotImplementedException();
+
+                    case OperationKind.Invalid:
+                        AddStatement(Visit(innerInitializer));
+                        return;
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(innerInitializer.Kind);
+                }
+            }
+
+            IOperation handleSimpleAssignment(ISimpleAssignmentOperation assignmentOperation)
+            {
+                // We spill all arguments to the lhs under the assumption that any of these could change after reading.
+                IOperation left = assignmentOperation.Target;
+                IOperation right;
+
+                switch (left.Kind)
+                {
+                    case OperationKind.FieldReference:
+                    case OperationKind.EventReference:
+                    case OperationKind.PropertyReference:
+                        var memberReference = (IMemberReferenceOperation)left;
+                        IPropertyReferenceOperation propertyReference = null;
+                        ImmutableArray<IArgumentOperation> propertyArguments = ImmutableArray<IArgumentOperation>.Empty;
+
+                        if (memberReference.Kind == OperationKind.PropertyReference &&
+                            !(propertyReference = (IPropertyReferenceOperation)memberReference).Arguments.IsEmpty)
+                        {
+                            var propertyArgumentsBuilder = ArrayBuilder<IArgumentOperation>.GetInstance();
```

Comment:
>.GetInstance(); [](start = 91, length = 15)

Can add propertyReference.Arguments.Count as a parameter to allocate a more efficient builder.

Reply:
Consider extracting an instance method to visit the argument and calculate `propertyArguments` with `SelectAsArray`. Then the `ArrayBuilder` can be removed, along with the check for `((IPropertyReferenceOperation)memberReference).Arguments.IsEmpty`.
```
var propertyArguments = propertyReference.Arguments.SelectAsArray(
    (arg, visitor) => visitor.HandleSimpleAssignmentArgument(arg),
    this);
```

Reply:
Never mind, the visit method has side effects so `SelectAsArray` does not feel appropriate.